custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -431,22 +428,40 @@ def _prepare_prediction_server_or_batch_pipeline(self, run_language):
 
     def _prepare_fit_pipeline(self, run_language):
 
-        if self.options.negative_class_label is None:
-            (
-                self.options.positive_class_label,
-                self.options.negative_class_label,
-            ) = possibly_intuit_order(
+        if self.target_type.value in TargetType.CLASSIFICATION.value:
+            posible_class_labels = possibly_intuit_order(
                 self.options.input,
                 self.options.target_csv,
                 self.options.target,
                 self.options.unsupervised,
             )
-        if self.options.unsupervised:
-            self._set_target_type(TargetType.ANOMALY)
-        elif self.options.negative_class_label is not None:
-            self._set_target_type(TargetType.BINARY)
-        else:
-            self._set_target_type(TargetType.REGRESSION)
+            if posible_class_labels is not None:
+                if (
+                    self.target_type == TargetType.BINARY
+                    and self.options.negative_class_label is None
+                ):
+                    if len(posible_class_labels) != 2:
```

Comment:
This if statement seems refinable.

Reply:
not sure what you want

Reply:
why can't you just do targettype==binary and len(class_labels) != 2

Reply:
I cleaned up the conditions, but left this aspect because I think it's valid to raise the exception where it's raised